openmc/universe.py

```diff
@@ -76,10 +76,17 @@ def __eq__(self, other):
             return False
         elif self.name != other.name:
             return False
-        elif self.cells != other.cells:
+        elif len(self.cells) != len(other.cells):
             return False
-        else:
-            return True
+
+        # Check each cell
+        for cell_id in self.cells:
+            if cell_id not in other.cells:
+                return False
+            if self.cells[cell_id] != other.cells[cell_id]:
+                return False
```

Comment:
Well tbh I cannot fully defend these changes and am happy to remove them. But during the process of debugging our issue with TREAT, I did a lot of comparisons of the OpenMC geometry before and after conversion to OpenMOC (i.e, the original compared to a version converted to OpenMOC and *back* to OpenMC). Although the geometries were identical (as evidenced by plots and tallied results), this equality method returned false for a few of the universes. The comparison of the dictionaries of cells returned false, while this new more elaborate comparison verified that the cells were identical. Any idea what would cause this behavior when comparing dictionaries of cells?